liby2/src/include/y2/Y2Namespace.h

```diff
@@ -132,11 +129,69 @@ class Y2Namespace {
 
     // pop all local variables from stack, uses SymbolEntry::pop()
     void popFromStack ();
-    
+
     // ensure that the namespace is initialized
     virtual void initialize ();
 
 };
 
+#define ERROR_NAMESPACE_NAME "_ErrorNamespace"
+class Y2ErrorNamespace : public Y2Namespace
+{
+public:
+
+  /**
+   * constructs new namespace in case when error happen
+   *
+   * @param summary short summary what error happen
+   * @param details can be longer details for error like e.g. backtrace
+   */
+  Y2ErrorNamespace(const std::string & summary, const std::string & details);
+
+  ~Y2ErrorNamespace();
+
+  /**
+   * shot summary what error happen
+   */
+  const std::string & summary() const;
+
+  /**
+   * longer details for error
+   */
+  const std::string & details() const;
+
+  const string name () const { return ERROR_NAMESPACE_NAME; }
+
+  const string filename () const { return "virtual_error_place"; }
+
+  /**
+   * creating funcition call for error namespace always fail
+   */
+  virtual Y2Function* createFunctionCall (const string name, constFunctionTypePtr type)
+  {
+    return NULL;
+  }
+
+  /**
+   * Does nothing for error namespace
+   */
+  YCPValue evaluate (bool cse = false);
+
+private:
+
+  string summary_;
+
+  string details_;
+};
+
+inline bool isErrorNamespace(const Y2Namespace * inst)
```

Comment:
We are introducing a special name and two helpers. The [use case is this](https://github.com/yast/yast-ruby-bindings/pull/181/files#diff-97f5abc054afe39f0bac0d02322bd999R95):
```cxx
if (isErrorNamespace(ns)) {
    Y2ErrorNamespace* ens = toErrorNamespace(ns);
    string message("Failed to load Module '");
    message = message + name + "' due to: " + ens->summary();
    ...
```

but IMHO it is clearer and simpler like this:

```cxx
Y2ErrorNamespace* ens = dynamic_cast<Y2ErrorNamespace*>(ns);
if (ens) {
    string message("Failed to load Module '");
    message = message + name + "' due to: " + ens->summary();
    ...
```
(or even `auto ens = ...`)

So I suggest to drop the helpers.